pjnath/src/pjnath/ice_session.c

```diff
@@ -3734,6 +3746,29 @@ PJ_DEF(pj_status_t) pj_ice_sess_on_rx_pkt(pj_ice_sess *ice,
 
         PJ_RACE_ME(5);
 
+        if (ice->opt.check_src_addr) {
+            for (i = 0; i < ice->rcand_cnt; ++i) {
+                if (ice->rcand[i].comp_id == comp_id && 
+                    ice->rcand[i].checked &&
+                    pj_sockaddr_cmp(src_addr, &ice->rcand[i].addr) == 0)
+                {
+                    break;
+                }
+            }
+            if (i == ice->rcand_cnt) {
+                char psrc_addr[PJ_INET6_ADDRSTRLEN] = {0};
+
+                if (pj_sockaddr_has_addr(src_addr)) {
+                    pj_sockaddr_print(src_addr, psrc_addr,
+                                      sizeof(psrc_addr), 3);
+                }
+                PJ_LOG(2, (ice->obj_name, "Ignoring incoming message for "
+                                          "component [%d], from src addr [%s]",
+                                          comp_id, psrc_addr));
```

Comment:
- Specify the reason of ignoring, e.g: s/"component [%d], from src addr [%s]"/"component %d because source address %s is unrecognized"? Note that `pj_sockaddr_print(..,3)` will also print bracket for IPv6.
- Won't log level 2 be too verbose for low level API, perhaps 4?